src/backend/gporca/libgpopt/src/base/CUtils.cpp

```diff
@@ -3276,6 +3277,54 @@ CUtils::FLogicalDML(COperator *pop)
 		   COperator::EopLogicalUpdate == op_id;
 }
 
+BOOL
+CUtils::FCTAS(COperator *pop)
+{
+	GPOS_ASSERT(NULL != pop);
+
+	if (COperator::EopLogicalInsert != pop->Eopid())
+	{
+		return false;
+	}
+
+	CLogicalInsert *popInsert = CLogicalInsert::PopConvert(pop);
+
+	return IMDId::EmdidGPDBCtas == popInsert->Ptabdesc()->MDId()->MdidType();
+}
+
+// recursively checks if the given expression has logical DML operator
+BOOL
+CUtils::FHasLogicalDML(CMemoryPool *mp, CExpression *pexpr)
+{
+	GPOS_ASSERT(NULL != mp);
+	GPOS_ASSERT(NULL != pexpr);
+
+	CDynamicPtrArray<CExpression, CleanupNULL> *exprsToCheck =
+		GPOS_NEW(mp) CDynamicPtrArray<CExpression, CleanupNULL>(mp);
+	exprsToCheck->Append(pexpr);
+
+	while (exprsToCheck->Size() > 0)
+	{
+		CExpression *pexprCurrent = (*exprsToCheck)[exprsToCheck->Size() - 1];
+		exprsToCheck->RemoveLast();
```

Comment:
`RemoveLast()` should 
```
// return the last element of the array and at the same time remove it from the array
```
so maybe
```suggestion
		CExpression *pexprCurrent = exprsToCheck->RemoveLast();
```
?

Reply:
done